From b65dfb6326b6f27967ed904d799be341b6fcfc27 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Mon, 24 Aug 2026 01:14:50 +0200
Subject: [PATCH] mmc: Don't use card_busy() shortcut for MMC_BUSY_ERASE

On mmc-sunxi card_busy reports busy=0 shortly after MMC_BUSY_ERASE
despite CMD13 reporting busy at the same time. (shortly after resume)

Don't use card_busy() shortcut to check for erase command busy status.

Signed-of-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/mmc/core/mmc_ops.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/core/mmc_ops.c b/drivers/mmc/core/mmc_ops.c
index a952cc8265af..4a24aca097c1 100644
--- a/drivers/mmc/core/mmc_ops.c
+++ b/drivers/mmc/core/mmc_ops.c
@@ -472,7 +472,8 @@ static int mmc_busy_cb(void *cb_data, bool *busy)
 	u32 status = 0;
 	int err;
 
-	if (data->busy_cmd != MMC_BUSY_IO && host->ops->card_busy) {
+	if (data->busy_cmd != MMC_BUSY_IO && data->busy_cmd != MMC_BUSY_ERASE &&
+	    host->ops->card_busy) {
 		*busy = host->ops->card_busy(host);
 		return 0;
 	}
-- 
2.43.0

